bin/docserv-createconfig

```diff
@@ -25,42 +25,120 @@ Git repositories.
+        url = url.replace(match.group(1), "https://")
+    return url
+
+def get_version(name):
+    """Returns version from product name
+    Example:
+    >>> get_version("SLE12SP4")
+    '12 SP4'
+
+    :param name: string with a product name
+    :type name: str
+
+    :return: the version of a product
+    :rtype: str
+    """
+    # example: ('SLE')('12')('SP4')
+    version_regex = r'(^[a-zA-Z_]+)([0-9.]+)?([a-zA-Z0-9.]+)?'
+    match = re.match(version_regex, name)
+    if match.group(2):
+            # substitute 'SLE' with '', removing product name
+            version = re.sub(r'(^[a-zA-Z_]+)', "", name)
+            if match.group(3):
+                # replace '12' with '12 ', adding space
+                version = version.replace(match.group(2), match.group(2) + " ")
+            # returning '12 SP4'
+            return version
+    return name
+
+
+def prettify(tree, indent="  "):
+    """Return a pretty-printed XML string for the Element.
+
+    :param tree: The XML tree
+    :type tree: :class:`xml.etree.ElementTree.ElementTree`
+
+    :param indent: string used for indentation (usually spaces)
+    :type indent: str
+
+    :return: the indented string
+    :rtype: str | byte
+    """
+    from xml.dom import minidom
+    from xml.etree import ElementTree
+    rough_string = ElementTree.tostring(tree.getroot(), encoding='UTF-8')
+    reparsed = minidom.parseString(rough_string)
+    return reparsed.toprettyxml(indent=indent)
+
+
+def generate_config(languages, git_url ,branches, version="0.9.10", contact=None):
     """Generate the configuration file.
 
     :param language: The allowed languages
     :type languages: list
 
-    :param branches:
-    :type branches:
+    :param branches: The branch name and their DC files, for example:
+               {'develop': ['DC-FOO-all', 'DC-FOO-install-quick'], ... }
+    :type branches: dict
 
     :param str schemaversion: the schema version to use
     :param str contact: the contact email address
     """
-    root = etree.Element("product", productid="FIXME", schemaversion=version)
-    etree.SubElement(root, "name").text = "FIXME"
-    etree.SubElement(root, "shortname").text = "FIXME"
+
+    root = etree.Element("product", productid=FIXME, schemaversion=version)
+    etree.SubElement(root, "git", remote=git_url)
+    etree.SubElement(root, "name").text = FIXME
+    etree.SubElement(root, "shortname").text = FIXME
     maintainers = etree.SubElement(root, "maintainers")
-    etree.SubElement(maintainers, "contact").text = "FIXME" if not contact else contact
+    etree.SubElement(maintainers, "contact").text = FIXME if not contact else contact
 
     for language in languages:
-        description = etree.SubElement(root, "desc", default="0", lang=language)
-        etree.SubElement(description, "p").text = "FIXME"
+        description = etree.SubElement(root, "desc",
+                                       default="0" if language != 'en-us' else "1", lang=language)
```

Comment:
This is taken from https://github.com/openSUSE/docserv/pull/108/files#r273770876 by Stefan:

----

This version is more "clever" than the dumb original but really not improving on the situation much:

* you are always picking en-us as the default language
* the default language needs to be the first language listed in the XML file

Issues would arise when ...

 * the user gives you a list like: de-de,fr-fr,en-us --> The result would be an invalid config file because the default language is not sorted first.
* the user gives you a list without en-us in it at all --> Again, invalid config file.

It would imo be a much better idea to ask the user what default language they want and then compare to the list of languages.

Reply:
Idea: Revert lines 122+123 back to the original ones. Then make sure you add the following change in `parse_cli()` function:

```diff
diff --git i/bin/docserv-createconfig w/bin/docserv-createconfig
index f4be738..ce6aaec 100755
--- i/bin/docserv-createconfig
+++ w/bin/docserv-createconfig
@@ -186,6 +186,9 @@ def parse_cli(cliargs=None):
                      "Expected one or more of these:\n"
                      "{}".format(args.languages, ", ".join(args.LANGUAGES))
                      )
+    # Push en-us at first position:
+    args.langs.remove('en-us')
+    args.langs.insert(0, 'en-us')
```